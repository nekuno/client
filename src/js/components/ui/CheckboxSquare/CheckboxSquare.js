import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './CheckboxSquare.scss';
import SquareIcon from "../SquareIcon/SquareIcon";
import RoundedIcon from "../RoundedIcon/RoundedIcon";

export default class CheckboxSquare extends Component {

    static propTypes = {
        onClickHandler : PropTypes.func,
        checked        : PropTypes.bool.isRequired,
        value          : PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);

        this.state = {
            checked: props.checked,
        };
    }

    handleClick() {
        this.setState({
            checked : !this.state.checked,
        });
        this.props.onClickHandler(!this.props.checked, this.props.value);
    }

    render() {
        const {children, checked} = this.props;

        return (
            this.state.checked ?
                <SquareIcon background={"#928bff"} icon={'check'} size={'xx-small'} color={'#ffffff'} onClickHandler={this.handleClick}/>
                :
                <SquareIcon background={"#ffffff"} border={'1px solid #c5d0de'} size={'xx-small'} color={'#ffffff'} onClickHandler={this.handleClick}/>
        );
    }
}