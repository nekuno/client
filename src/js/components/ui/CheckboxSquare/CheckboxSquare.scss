@import '../../../../scss/_partials/colors';

.checkboxsquare {
  background: #fff;
  font-size: 15px;
  padding: 25px;
}