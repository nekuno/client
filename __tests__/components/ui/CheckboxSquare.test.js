import React from 'react';
import CheckboxSquare from '../../../src/js/components/ui/CheckboxSquare/CheckboxSquare.js';
import styles from '../../../src/js/components/ui/CheckboxSquare/CheckboxSquare.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test CheckboxSquare component', () => {
    const mockFn = jest.fn();
    const checkboxsquare = shallow(<CheckboxSquare onClickHandler={mockFn}>Lorem ipsum</CheckboxSquare>);

    it('should be defined', () => {
        expect(checkboxsquare).toBeDefined();
    });
    it('should render correctly', () => {
        expect(checkboxsquare).toMatchSnapshot();
    });
    it('CheckboxSquare click event', () => {
        checkboxsquare.find(`.${styles.checkboxsquare}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
