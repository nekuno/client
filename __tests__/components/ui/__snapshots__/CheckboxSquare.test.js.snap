// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckboxSquare component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckboxSquare
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </CheckboxSquare>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem ipsum",
      "className": "checkboxsquare",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem ipsum",
        "className": "checkboxsquare",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
